src/nxt_tstr.c

```diff
@@ -214,9 +208,7 @@ nxt_tstr_str(nxt_tstr_t *tstr, nxt_str_t *str)
 {
     *str = tstr->str;
 
-    if (tstr->flags & NXT_TSTR_STRZ) {
-        str->length--;
-    }
+    str->length--;  // Hide the terminating NUL.
```

Comment:
Not sure I'm keen on these comments, particularly the term 'Hide'.

We didn't need the comment before. I would suggest to remove it and do

```c
tstr->str.length = str->length;

tstr->str.start = nxt_mp_nget(state->pool, str->length + 1);
```
Then you just _don't_ need (I guess, it seems to happen way off somewhere else in the code).
```c
str->length--;
```
I think this also improves the code... and renders the need for any comment moot.

Reply:
I'm neither.

The reason I wrote them was to remind me that I needed to dig more in how this tstr is implemented, so that I never add that +1 in the length in the first place.  I think I've managed to do it.  And maybe this fixes the NJS build (I couldn't test it).  See the v4 I'll push in a moment with an amend commit.